packages/firestore/src/remote/remote_store.ts

```diff
@@ -185,6 +185,9 @@ export class RemoteStore {
     const didChange = this.watchStreamOnlineState !== onlineState;
     this.watchStreamOnlineState = onlineState;
     if (didChange) {
+      if (onlineState == OnlineState.Failed) {
```

Comment:
Per my comment on the Android version, I think we should only log this once, and only if we've never succeeded.

Reply:
Done. Added a variable to track the logic: shouldWarnOffline, which is initialized to true and changed to false if when ever online and when ever offline is logged.